lua/py-requirements/requirements.lua

```diff
@@ -32,6 +32,10 @@ local M = {}
 ---@param line string
 ---@return PythonModule|nil
 function M.parse_module(line_number, line)
+    if string.match(line, '^%s*--hash') then
```

Comment:
The line by line parsing is an artifact of before I moved to using tree-sitter to pull information out of the file. 
I hadn't considered also just parsing the whole thing at once but I really like the idea, thank you!
Totally good leaving this in place for now as I don't know when I'll make the larger change.